synapse/api/auth.py

```diff
@@ -87,11 +87,13 @@ def __init__(self, hs):
     async def check_from_context(
         self, room_version: str, event, context, do_sig_check=True
     ):
-        prev_state_ids = await context.get_prev_state_ids()
-        auth_events_ids = self.compute_auth_events(
-            event, prev_state_ids, for_verification=True
-        )
-        auth_events = await self.store.get_events(auth_events_ids)
+        auth_event_ids = event.auth_event_ids()
```

Comment:
Use the `auth_event_ids` from the event if available. Otherwise go down the route where we derive from the `prev_state_ids`

Reply:
Err, event.auth_event_ids() should never return a None afaik?

Reply:
hmm, can we simplify and always use `event.auth_event_ids()`?

It could be `None` if no `auth_events` were provided to the event when it was created.

In [`builder.build(...)`](https://github.com/matrix-org/synapse/blob/d9bd181a3fd1431af784c525e7fe552e0bcad48e/synapse/events/builder.py#L103), we do fallback to populating `auth_events` if `None` is provided. In the [federation stuff](https://github.com/matrix-org/synapse/blob/d9bd181a3fd1431af784c525e7fe552e0bcad48e/synapse/replication/http/federation.py#L113-L120), we also create events and I don't see any validation that `auth_events` is in the dict before we `make_event_from_dict`.

Reply:
I think that's mostly because we suck at validation in Synapse, an event with a `None` auth_events` is just invalid (and would fail all the auth checks on remote servers)

Reply:
> can we simplify and always use `event.auth_event_ids()`?

@erikjohnston  ^^

Maybe it's a case of removing and see if all of the tests pass. Works fine in my Complement tests.

Reply:
Yeah I think so